src/main/java/io/papermc/hangar/service/api/ProjectsApiService.java

```diff
@@ -81,8 +98,45 @@ public Project getProject(String author, String slug) {
             ordering = orderingFirstHalf + relevance;
         }
 
+        if(paginationSortValue != null && !paginationSortValue.isBlank()){
+            String relevance = "ASC";
+            String orderingFirstHalf;
+
+            switch (sort) {
+                case STARS -> {
+                    orderingFirstHalf = "hp.stars ";
+                    relevance = "DESC";
+                }
+                case DOWNLOADS -> {
+                    orderingFirstHalf = "hp.downloads ";
+                    relevance = "DESC";
+                }
+                case VIEWS -> {
+                    orderingFirstHalf = "hp.views ";
+                    relevance = "DESC";
+                }
+                //case NEWEST: orderingFirstHalf ="((EXTRACT(EPOCH FROM hp.created_at) - 1609459200) / 86400) *"; break;
+                case NEWEST -> {
+                    orderingFirstHalf = "newest_double ";
+                    relevance = "DESC";
+                }
+                case UPDATED -> {
+                    orderingFirstHalf = "last_updated_double ";
+                    relevance = "DESC";
+                }
+                case ONLY_RELEVANCE -> orderingFirstHalf = "";
+                case RECENT_DOWNLOADS -> orderingFirstHalf = "hp.recent_views ";
+                case RECENT_VIEWS -> orderingFirstHalf = "hp.recent_downloads ";
+                default -> orderingFirstHalf = " "; // Just in case and so that the ide doesn't complain
+            }
+            ordering = orderingFirstHalf + relevance;
+
+        }
+
```

Comment:
This should be implemented in a similar way to fitlers, with a registry and using annotation to define valid sorters for api requests. This abstracts sorting to be valid for more things than just this one api request.